pkg/internal/serialization.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"github.com/go-openapi/jsonreference"
+)
+
+func JSONRefFromMap(v map[string]interface{}) (*jsonreference.Ref, error) {
```

Comment:
At first I was wondering why this returns a pointer to ref, but it's because you want to differentiate empty ref from missing ref. I don't think we need to differentiate? See my comment below.